src/coaster_gui.cpp

```diff
@@ -17,10 +17,63 @@
 #include "viewport.h"
 #include "map.h"
 #include "gui_sprites.h"
+#include "entity_gui.h"
+
+/** Window to prompt for removing a roller coaster. */
+class CoasterRemoveWindow : public EntityRemoveWindow  {
+	public:
+		CoasterRemoveWindow(CoasterInstance *ci);
+
+		void OnClick(WidgetNumber number, const Point16 &pos) override;
+		void SetWidgetStringParameters(WidgetNumber wid_num) const override;
+
+	private:
+		CoasterInstance *ci; ///< Roller coaster instance to remove.
+};
+
+/**
+ * Constructor of the roller coaster remove window.
+ * @param ci Roller coaster instance to remove.
+ */
+CoasterRemoveWindow::CoasterRemoveWindow(CoasterInstance *ci) : EntityRemoveWindow(WC_COASTER_REMOVE, ci->GetIndex())
+{
+	this->ci = ci;
+}
+
+void CoasterRemoveWindow::OnClick(WidgetNumber number, const Point16 &pos)
+{
+	if (number == ERW_YES) {
+		Window *w = GetWindowByType(WC_COASTER_MANAGER, this->ci->GetIndex());
+		delete w;
+		_rides_manager.DeleteInstance(this->ci->GetIndex());
+	}
+	delete this;
+}
+
+void CoasterRemoveWindow::SetWidgetStringParameters(WidgetNumber wid_num) const
+{
+	switch (wid_num) {
+		case ERW_MESSAGE:
+			_str_params.SetUint8(1, (uint8 *)this->ci->name);
+			break;
+	}
```

Comment:
Is it worth bothering with a one case switch such as this? Especially without a default - some compilers will complain that you haven't covered all options